(self.webpackChunk_N_E=self.webpackChunk_N_E||[]).push([[4140],{50086:function(e,r,n){(window.__NEXT_P=window.__NEXT_P||[]).push(["/quizzes/javascript-debugging-testing/handling-errors-with-try-catch-finally/what-will-be-logged-as-the-error-name-in-this-code",function(){return n(73041)}])},16833:function(e,r,n){"use strict";n.d(r,{Z:function(){return i}});var t=n(85893),o=n(67294),s=e=>{let{adClient:r="ca-pub-7544108447932228",adSlot:n="6753219060",className:s}=e,i=(0,o.useRef)(null),a=(0,o.useRef)(!1),[l,c]=(0,o.useState)(!1);return((0,o.useEffect)(()=>(c(!0),()=>{let e=document.getElementById("__next");e&&e.removeAttribute("style")}),[]),(0,o.useEffect)(()=>{if(l){if(document.querySelector('script[src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"]'))i.current&&!a.current&&((window.adsbygoogle=window.adsbygoogle||[]).push({}),a.current=!0);else{let e=document.createElement("script");e.src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-7544108447932228",e.async=!0,e.crossOrigin="anonymous",document.body.appendChild(e),e.onload=()=>{i.current&&!a.current&&((window.adsbygoogle=window.adsbygoogle||[]).push({}),a.current=!0)}}}},[l]),l)?(0,t.jsx)("ins",{ref:i,className:"adsbygoogle ".concat(s),style:{display:"block"},"data-ad-client":r,"data-ad-slot":n}):null};function i(e){return class extends o.Component{componentDidMount(){console.log("Wrapped Component")}render(){return(0,t.jsx)(t.Fragment,{children:(0,t.jsxs)("div",{className:" md:grid grid-cols-[20%_60%_20%]  w-[95%] ",children:[(0,t.jsx)("div",{children:(0,t.jsx)(s,{adSlot:"6753219060",className:"h-[100px] md:h-full"})}),(0,t.jsx)("div",{children:(0,t.jsx)(e,{...this.props})}),(0,t.jsx)("div",{children:(0,t.jsx)(s,{adSlot:"6208304795",className:"h-[300px] md:h-full"})})]})})}constructor(e){super(e)}}}},73041:function(e,r,n){"use strict";n.r(r),n.d(r,{__N_SSG:function(){return a}});var t=n(85893),o=n(1475),s=n(25190),i=n(16833),a=!0;r.default=(0,s.Z)((0,i.Z)(()=>(0,t.jsxs)(t.Fragment,{children:[(0,t.jsx)("h1",{className:"text-2xl dark:text-white text-center",children:"Handling Errors with try-catch-finally"}),(0,t.jsx)("div",{className:"flex justify-center w-full p-5 md:p-0 pt-0",children:(0,t.jsx)(o.Y3,{activeQuestionIndex:0,totalQuestions:20,index:0,question:"What will be logged as the error name in this code?",options:["SyntaxError","ReferenceError","TypeError","NullError"],code:"try {\n  const obj = null;\n  obj.property;\n} catch (error) {\n  console.log(error.name);\n}",nextQuestionPath:"what-is-the-purpose-of-rethrowing-an-error-with-additional-information",correctAnswer:3,explanation:"This code will log 'TypeError': 1) Attempting to access a property of null triggers a TypeError, 2) TypeError occurs when an operation is performed on a value of the wrong type, 3) Null is not an object and has no properties to access, 4) The error.name property specifically identifies the error type, 5) This is one of JavaScript's built-in error types, 6) TypeError is commonly encountered when dealing with undefined or null values, 7) Understanding error types helps in proper error handling, 8) This pattern is useful for debugging null reference issues.",id:1141,quizId:49},0)})]})))}},function(e){e.O(0,[41966,25675,82893,92888,49774,40179],function(){return e(e.s=50086)}),_N_E=e.O()}]);